### PR TITLE
Migrate dependency to terraform 0.11.13 (Closes #181)

### LINKED ISSUE #181: Migrate github.com/hashicorp/terraform dependency from v0.10.0 to v0.11.11
Terraform v0.10.0 was release in August 2, 2017. This PR is to bump this dependency (and it's direct dependencies to latest stable v0.11.11 (December 14, 2018).

Time gap between packages is quite lengthy. I've reviewed changelog and found related work done on SDK side listed below

0.11.0 (November 16, 2017):
* helper/schema: Loosen validation for 'id' field (#16456)

0.11.4 (March 15, 2018):
* helper/schema: Prevent crash on removal of computed field in CustomizeDiff (#17261)
* helper/schema: Allow ResourceDiff.ForceNew on nested fields (avoid crash) (#17463)
* helper/schema: Allow TypeMap to have a *schema.Schema as its Elem, for consistency with TypeSet and TypeList (#17097)
* helper/validation: Add ValidateRFC3339TimeString function (#17484)

0.11.6 (April 5, 2018):
* helper/schema: Attribute value is no longer included in error message when ConflictsWith keys are used together. (#17738)

0.11.7 (April 10, 2018):
* helper/schema: Invoking ForceNew on a key being removed from config during diff customization now correctly exposes that key as being removed in the updated diff. This prevents diff mismatches under certain circumstances. (#17811)
* helper/schema: Invoking ForceNew during diff customization on its own no longer writes any new data to the diff. This prevents writing of new nil to zero value diffs for sub-fields of complex lists and sets where a diff did not exist before. (#17811)

### PR DESCRIPTION
**Pull request by Didainius**
As part of getting the provider codebase up to date this PR upgrades terraform provider dependencies to ~~v0.11.11~~ v0.11.13 (latest stable).

Such actions were taken:
* Cleanup all indirect dependencies from go.mod
* Bump terraform dependency to ~~v0.11.11~~ v0.11.13 in go.mod
* Perform `go mod tidy -v` so that `go.mod` and `go.sum` have indirect dependencies required by ~~v0.11.11~~ v0.11.13 pulled in
* Rebuild vendor directory with `go mod vendor`
* Ensuring test suite is running:
  * Issues were spotted with `TestAccVcdNetworkDirect` and solved in aa7c3a5. The `ExpectError` wasn't doing any good in this test but wasn't causing an error due to Terraform handles `ExpectError` differently in old and current version - before it used to be https://github.com/hashicorp/terraform/blob/v0.10.0/helper/resource/testing.go#L421-L426, now it is https://github.com/hashicorp/terraform/blob/v0.11.11/helper/resource/testing.go#L516-L524. __Main difference__ is that in v0.10.0 Terraform didn't check if `ExpectError` existed when an error didn't occur, now it does and causes the suite to fail because in fact there is no error

### CODE REVIEW

**Comment by Didainius** [comment]
> > +476,025 −14,934
> 
> Why there are so many _additions_ to the vendor folder?

I can only speculate that dependency tree of terraform has grown. If you do the below actions on my branch `migrate_to_terraform_0.11.11` proof of it is there:
```
$ rm -rf vendor # remove vendor directory
$ cat go.mod #(after cleaning up all // indirect dependencies)
module github.com/terraform-providers/terraform-provider-vcd/v2

go 1.12

require (
        github.com/hashicorp/terraform v0.11.11
        github.com/vmware/go-vcloud-director/v2 v2.1.0-alpha.2
)

$ go mod tidy # will pull in all //indirect dependencies as needed for terraform
$ cat go.mod
module github.com/terraform-providers/terraform-provider-vcd/v2

go 1.12

require (
        github.com/apparentlymart/go-cidr v1.0.0 // indirect
        github.com/blang/semver v3.5.1+incompatible // indirect
        github.com/hashicorp/go-getter v1.2.0 // indirect
        github.com/hashicorp/go-hclog v0.8.0 // indirect
        github.com/hashicorp/go-plugin v0.0.0-20190220160451-3f118e8ee104 // indirect
        github.com/hashicorp/go-uuid v1.0.1 // indirect
        github.com/hashicorp/hcl v1.0.0 // indirect
        github.com/hashicorp/hcl2 v0.0.0-20190305174554-fdf8e232b64f // indirect
        github.com/hashicorp/hil v0.0.0-20190212132231-97b3a9cdfa93 // indirect
        github.com/hashicorp/logutils v1.0.0 // indirect
        github.com/hashicorp/terraform v0.11.11
        github.com/mitchellh/cli v1.0.0 // indirect
        github.com/mitchellh/copystructure v1.0.0 // indirect
        github.com/mitchellh/go-homedir v1.1.0 // indirect
        github.com/mitchellh/hashstructure v1.0.0 // indirect
        github.com/vmware/go-vcloud-director/v2 v2.1.0-alpha.2
        github.com/zclconf/go-cty v0.0.0-20190212192503-19dda139b164 // indirect
        golang.org/x/crypto v0.0.0-20190228161510-8dd112bcdc25 // indirect
        golang.org/x/net v0.0.0-20190301231341-16b79f2e4e95 // indirect
)

$ go mod vendor # will rebuild the vendor directory
$ git status
On branch migrate_to_terraform_0.11.11
Your branch is up to date with 'origin/migrate_to_terraform_0.11.11'.

nothing to commit, working tree clean
```

The branch becomes exactly as it is in my PR.

**Comment by lvirbalas** [comment]
@appilon we'd like to ask you to review this Terraform upgrade, as we're doing it for the first time and have also switched to modules, which resulted in a lot of changes of the /vendor folder (which we are continuing to support per your request). Please double check our pulled dependencies. Also, please let us know when should we plan to remove vendor/ folder completely.

**Comment by appilon** [comment]
@lvirbalas I don't have any finite timelines for when we will stop vendoring, apologies i know it is a pain to see in the git history/pull requests. I would like to recommend upgrading to a reference of terraform based on v0.11.12. It supports modules which will reduce the amount of `indirect` requirements in this project's `go.mod` file. It will not reduce the amount of files that are vendored, however future versions of terraform will try to reduce the dependency story.

Just run
```
GO111MODULE=on go get github.com/hashicorp/terraform@sdk-v0.11-with-go-modules
GO111MODULE=on go mod tidy
GO111MODULE=on go mod vendor
```
There should be no SDK related changes between v0.11.11 and this version, it should be a safe upgrade.

**Comment by appilon** [comment]
I would also like to link another modules PR to illustrate a few other minor changes needed in an effort to keep our providers consistent:

https://github.com/terraform-providers/terraform-provider-icinga2/pull/29/files

1) We would like to pin the version of go into a `goenv` file: `.go-version`. We had been pinning to go1.11.5, I see in this PR you have moved ahead to go1.12, that should be fine, could you encode a `.go-version` file.

2) I noticed your travis setup test with both modules on and off, we have not made that a requirement for any providers, just want to make sure travis works consistently. To achieve that consistency we would recommend setting `GO111MODULE=on` and `GOFLAGS=-mod=vendor` that will tell Go to take the projects `vendor/` folder as the source of dependencies.

Thank you for your time and contributions!

**Comment by Didainius** [comment]
Thanks @appilon for your comments.
> I would like to recommend upgrading to a reference of terraform based on v0.11.12. It supports modules which will reduce the amount of `indirect` requirements in this project's `go.mod`

I noticed that this points to `v0.11.12-beta1.0.20190227065421-fc531f54a878` I guess we can stay with just a few more //indirect imports in go.mod and bump up to v0.11.12 when it's released as stable. 

>     1. We would like to pin the version of go into a `goenv` file: `.go-version`. We had been pinning to go1.11.5, I see in this PR you have moved ahead to go1.12, that should be fine, could you encode a `.go-version` file.

Done



>     2. I noticed your travis setup test with both modules on and off, we have not made that a requirement for any providers, just want to make sure travis works consistently. To achieve that consistency we would recommend setting `GO111MODULE=on` and `GOFLAGS=-mod=vendor` that will tell Go to take the projects `vendor/` folder as the source of dependencies.

Adjusted with still leaving two options and a check for `go.mod` and `vendor` being in sync for now.

**Comment by Didainius** [comment]
@appilon do you have any extra comments? I've bumped dependency to v0.11.13 and added .go-version.

File: .go-version [hunk 1]
```diff
@@ -0,0 +1,2 @@
+1.12
+
```
Comment (appilon) [inline .go-version]: might be 1 newline too many but I don't know if `goenv` cares

**Comment by appilon** [comment]
Looks good 👍

**Pull request merged**